tvb_epilepsy/base/disease_hypothesis.py

```diff
@@ -4,75 +4,231 @@
 """
 from collections import OrderedDict
 
-import numpy
+import numpy as np
 
-from tvb_epilepsy.base.h5_model import prepare_for_h5
-from tvb_epilepsy.base.utils import formal_repr
+from tvb_epilepsy.base.utils import initialize_logger, formal_repr, ensure_list, dicts_of_lists_to_lists_of_dicts, \
+                                    linear_index_to_coordinate_tuples
+from tvb_epilepsy.base.h5_model import convert_to_h5_model
+
+from tvb_epilepsy.base.model_configuration import ModelConfiguration
+from tvb_epilepsy.base.constants import FOLDER_FIGURES, VERY_LARGE_SIZE, FIG_FORMAT, SAVE_FLAG, SHOW_FLAG
+from tvb_epilepsy.base.plot_factory import plot_in_columns
+
+from model_vep import Connectivity
 
 # NOTES:
-#  For the moment a hypothesis concerns the excitability and/or epileptogenicity of each brain region.
-# TODO: Generate a richer disease hypothesis as a combination of 4 kinds of hypotheses:
-# a. the existing excitability and/or epileptogenicity one
-# b. a connectivity one leading to changes to the connectivity matrix, and/or to the K global coupling scaling parameter
-# c. one that changes some of the parameters a, b, d, yc, Iext1 that belong to the x1, z state variable (sub)system
-# d. one that changes other parameters that do not affect lsa, but will have an effect for simulations of >2D model
+#  For the moment a hypothesis concerns the excitability and/or epileptogenicity of each brain region,
+#  and/or scalings of specific connectivity weights.
+# TODO if needed in the future: Generate a richer disease hypothesis as a combination of hypotheses on other parameters.
+
+logger = initialize_logger(__name__)
+
 
 class DiseaseHypothesis(object):
-    def __init__(self, connectivity, disease_values, x0_indices=[], e_indices=[], propagation_indices=[],
-                 propagation_strenghts=[], hypo_type="Excitability", name=""):
-        self.type = hypo_type
+    def __init__(self, connectivity, excitability_hypothesis={}, epileptogenicity_hypothesis={},
+                 connectivity_hypothesis={}, propagation_indices=[], propagation_strenghts=[], name=""):
+
```

Comment:
This method now looks like a proper Factory for Hypothesis, and could be separated into an actual HypothesisFactory class.